src/legacy/ui/public/index_patterns/_index_pattern.js

```diff
@@ -269,6 +277,16 @@ export function IndexPatternProvider(Private, config, Promise, confirmModalPromi
         .then(() => this);
     }
 
+    migrate(newTitle) {
+      return savedObjectsClient.update(type, this.id, {
+        title: newTitle,
+        intervalName: null
+      }).then(({ attributes: { title, intervalName } }) => {
+        this.title = title;
+        this.intervalName = intervalName;
+      }).then(() => this);
```

Comment:
Maybe it's a 'nip' from my side but I don't understand why we need it  `.then(() => this);`
We can just return `this` from the previous `then`.

It may look a little worse, but we are not planning an additional Microtask.

Reply:
Yeah, that's totally true. I think I was just following the same style as above, but it could definitely be changed.